source/Halibut.Tests/Transport/ListeningServiceErrorConnectionStateFixture.cs

```diff
@@ -49,6 +50,28 @@ await using (var clientAndService = await clientAndServiceTestCase.CreateTestCas
                                  portForwarderRef.Value.EnterKillNewAndExistingConnectionsMode();
                              })
                              .Build(CancellationToken))
+            {
```

Comment:
It's not obvious from the diff, but this test had to be adjusted.

It was configured to fail during `WithDoSomeActionService`. But rather than create and call an `IDoSomeActionService`, it ended up creating an `echoService`. 

As a result, it would fail with the right exception, but it was going down a different code path and not actually hitting the path this test was intended to hit.